soccer/include/rj_vision_filter/VisionFilter.hpp

```diff
@@ -63,15 +89,62 @@ class VisionFilter {
      */
     void GetFrames();
 
-    std::thread worker;
+    /**
+     * @brief Returns the team angle (It's horrible, but it's only temporary
+     * until VisionFilter gets refactored).
+     * @return The team angle.
+     */
+    [[nodiscard]] double TeamAngle() const {
+        const bool defend_plus_x = config_client_.gameSettings().defend_plus_x;
+        return defend_plus_x ? -M_PI_2 : M_PI_2;
+    }
+
+    /**
+     * @brief Returns the transform from the world to the team.
+     * @return The transform from the world to the team frame.
+     */
+    [[nodiscard]] Geometry2d::TransformMatrix WorldToTeam() const {
+        Geometry2d::TransformMatrix world_to_team =
+            Geometry2d::TransformMatrix::translate(
+                0, config_client_.fieldDimensions().length / 2.0f);
+        world_to_team *= Geometry2d::TransformMatrix::rotate(
+            static_cast<float>(TeamAngle()));
+        return world_to_team;
+    }
 
-    std::mutex worldLock;
+    mutable std::mutex world_mutex;
     World world;
 
-    std::atomic_bool threadEnd{};
-
     std::vector<CameraFrame> frameBuffer{};
-    Context* context_;
-    std::unique_ptr<ros2_temp::DetectionFrameSub> detection_frame_sub_;
-    RJ::Time last_update_time_;
-};
\ No newline at end of file
+
+    /**
+     * @brief Timestamp of the last received vision message. Has value
+     * std::nullopt if we haven't received any vision messages yet.
+     */
+    std::optional<RJ::Time> last_update_time_;
+
+    rclcpp::CallbackGroup::SharedPtr publish_timer_callback_group_;
```

Comment:
Man, trying to read this section on mobile is rough lol. It's super dense in terms of Ros objects. Not sure if there's a way to fix it though

Reply:
Done.